src/ai/defaultai.cc

```diff
@@ -2211,6 +2368,32 @@ void DefaultAI::update_productionsite_stats() {
 	}
 }
 
+unsigned DefaultAI::find_immovables_nearby(
```

Comment:
this funtion should be usable for detection of supportinmg buildings nearby in the upgrade algorithm section.

Reply:
forget this comment Ifound a differnet solution